teos/src/main.rs

```diff
@@ -186,14 +183,50 @@ async fn main() {
     // Load last known block from DB if found. Poll it from Bitcoind otherwise.
     let last_known_block = dbm.lock().unwrap().load_last_known_block();
     let tip = if let Some(block_hash) = last_known_block {
-        derefed
+        let mut last_known_header = derefed
             .get_header(&block_hash, None)
             .await
             .unwrap()
             .validate(block_hash)
-            .unwrap()
+            .unwrap();
+
+        log::info!(
+            "Last known block: {}",
+            last_known_header.header.block_hash()
+        );
+
+        // If we are running in pruned mode some data may be missing (if we happen to have been offline for a wile)
+        if let Some(prune_height) = rpc.get_blockchain_info().unwrap().prune_height {
+            if last_known_header.height < prune_height as u32 {
```

Comment:
We would still fail if some of the 100 blocks that proceeds the last known one are pruned.

The check should be: `last_known_header.height - 100 < prune_height as u32`

Reply:
`<=`, but you're right

Reply:
Oh, NVM, I was thinking about the positive case (`>=`)